pkg/serializer/internal/metrics/events.go

```diff
@@ -40,7 +36,10 @@ var (
 )
 
 // Events represents a list of events ready to be serialize
-type Events []*event.Event
+type Events struct {
+	EventsArr []*event.Event
+	Hostname  string
```

Comment:
Would it be simpler/less code to store `Hostname` on `eventsBySourceTypeMarshaler`, and add it as an argument to [`CreateSingleMarshaler`](https://github.com/DataDog/datadog-agent/blob/43053f3ea68b19a23840739a07c9f3d9c44ae2d7/pkg/serializer/internal/metrics/events.go#L237) and `CreateMarshalersBySourceType`?

Both are called from `serializer.go` which has a copy of the hostname already.

Reply:
Setting Hostname on 'Events` is because of the `MarshalJSON` method.  We can't add arguments to the method or change the structure because of the API contract.